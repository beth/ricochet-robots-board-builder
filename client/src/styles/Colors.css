:local .colors {
  display: grid;
  grid-template-rows: repeat(2, 1fr);
  grid-template-columns: repeat(2, 1fr);
  align-content: center;
  justify-items: center;
}
