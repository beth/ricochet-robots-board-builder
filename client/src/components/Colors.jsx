import React from 'react';
import PropTypes from 'prop-types';
import ColorContainer from '../containers/ColorContainer';
import styles from '../styles/Colors.css';

const Colors = ({ colors, row, col }) => (
  <div className={styles.colors}>
    { colors.map(c => <ColorContainer color={c} row={row} col={col} key={c} />) }
  </div>
);

Colors.propTypes = {
  colors: PropTypes.arrayOf(PropTypes.string).isRequired,
  row: PropTypes.number.isRequired,
  col: PropTypes.number.isRequired,
};

export default Colors;
