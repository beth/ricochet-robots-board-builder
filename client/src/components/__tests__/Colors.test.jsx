import React from 'react';
import renderer from 'react-test-renderer';
import Colors from '../Colors';

jest.mock('../../containers/ColorContainer', () => 'ColorContainer');

describe('Colors Component', () => {
  it('renders four ColorContainers when given four colors', () => {
    const domTree = renderer.create(<Colors
      colors={['RED', 'BLUE', 'YELLOW', 'GREEN']}
      row={8}
      col={8}
    />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it('renders two ColorContainers when given two colors', () => {
    const domTree = renderer.create(<Colors
      colors={['RED', 'GREEN']}
      row={8}
      col={8}
    />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
