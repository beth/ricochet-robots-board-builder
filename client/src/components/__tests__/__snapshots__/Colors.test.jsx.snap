// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors Component renders four ColorContainers when given four colors 1`] = `
<div
  className="colors"
>
  <ColorContainer
    col={8}
    color="RED"
    row={8}
  />
  <ColorContainer
    col={8}
    color="BLUE"
    row={8}
  />
  <ColorContainer
    col={8}
    color="YELLOW"
    row={8}
  />
  <ColorContainer
    col={8}
    color="GREEN"
    row={8}
  />
</div>
`;

exports[`Colors Component renders two ColorContainers when given two colors 1`] = `
<div
  className="colors"
>
  <ColorContainer
    col={8}
    color="RED"
    row={8}
  />
  <ColorContainer
    col={8}
    color="GREEN"
    row={8}
  />
</div>
`;
